src/app/services/indicator.service.ts

```diff
@@ -24,6 +25,16 @@ export class IndicatorService {
         // Generate query params
         const searchParams: URLSearchParams = new URLSearchParams();
         const optParams = options.params;
+
+        // append extra parameters for threshold indicators
+        if (options.indicator.thresholdIndicator) {
+            // abort request if chart is in flux (these parameters are required)
+            if (!optParams.threshold) return Observable.of({url: ''});
```

Comment:
Given the extra checks that TypeScript gives us, I think this is a dangerous thing to return. While the method technically returns an `Observable<any>`, really its expected to return something like `Observable<IndicatorDataResponse>`, which is a large convoluted type that we haven't actually defined anywhere. `Observable.of({url: ''})` wouldn't match that nebulous type that we haven't defined for sure. It's probably better to raise a specific error with `Observable.throw()` and have the calling code handle this specific problem.